Makefile

```diff
@@ -102,7 +102,7 @@ vet: ## Run go vet against code.
 	go vet ./...
 
 .PHONY: test
-test: manifests generate fmt vet envtest ## Run tests.
+test: manifests generate fmt vet ensure-generate-is-noop envtest ## Run tests.
```

Comment:
Why `ensure-generate-is-noop` is needed for running the unit tests?

Reply:
Why not? that we you just call `make test` and get a result if your state is ready to commit.

Reply:
Because if you are doing quick changes to test stuff, you'll need to commit the results all the time in order to really run the rules from `make test`.
For me, it would make more sense to create a git hook instead of forcing the `ensure-generate-is-noop` rule here.

Reply:
I agree with @iblancasa here.  Having to commit changes just to run the unit-tests is a nuisance.

Reply:
I followed here the [OpenTelemetry-Operator](https://github.com/open-telemetry/opentelemetry-operator/blob/v0.59.0/Makefile#L86) approach. According to my understanding the sense behind it is that `make test` calls all needed tests. To test parts, you can then fall back on e.g. manual cli commands. But should be a big deal to register it as an extra ci step.

Git hooks would prevent you from creating an extra generation commit. Which makes the review quite painful.

Reply:
If the problem is related to the CI, just add this dependency to the `ci` Make target. 

> Git hooks would prevent you from creating an extra generation commit. Which makes the review quite painful.

I don't think so. It will prevent you of making a commit of non checked in files.
From what I understand, you don't like this approach because you will have one commit for your changes + another one for committing the offending files for `ensure-generate-is-noop`, right? But if you force to commit them each time we want to run `make test`, we will have a commit for each change done affecting the tests and the files generation.

I think the correct solution should be add this dependency to the `ci` target.